File: Backend/cart-service/src/main/java/com/shoppingcart/Cart/Model/Product.java
package com.shoppingcart.Cart.Model;

import java.util.Map;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "Products")
public class Product {
	
	@Transient
    public static final String SEQUENCE_NAME = "products_sequence";
	
	@Id
	private int productId;
	private String productType;
	private String productName;
	private String category;
	private Map<Integer,Double> rating;
	private Map<Integer,String> review;
	private String image1;
	private String image2;
	private String image3;
	private String image4;
	private double price;
	private String description;
	private Map<String,String> specification;
	
	
	
	
	

}

File: frontend/Shopping-Cart/src/app/models/product.ts
export interface Product {
  productId: number;
  productType: string;
  productName: string;
  category: string;
  rating: Map<number,number>;
  review: Map<number,string>;
  image1 : string;
  image2 : string;
  image3 : string;
  image4 : string;
  price:number;
  description:string;
  specification: key<string,string>;
  cost:String;
}

export interface key<K, V> {
  key: K
  value: V
}
